pandas/core/arrays/interval.py

```diff
@@ -0,0 +1,1049 @@
+        IntervalArray
+        """
+        closed = set(interval.closed for interval in to_concat)
+        if len(closed) != 1:
+            raise ValueError("Intervals must all be closed on the same side.")
+        closed = closed.pop()
+
+        left = np.concatenate([interval.left for interval in to_concat])
+        right = np.concatenate([interval.right for interval in to_concat])
+        return cls._simple_new(left, right, closed=closed, copy=False)
+
+    def _shallow_copy(self, left=None, right=None, closed=None):
+        """
+        Return a new IntervalArray with the replacement attributes
+
+        Parameters
+        ----------
+        left : array-like
+            Values to be used for the left-side of the the intervals.
+            If None, the existing left and right values will be used.
+
+        right : array-like
+            Values to be used for the right-side of the the intervals.
+            If None and left is IntervalArray-like, the left and right
+            of the IntervalArray-like will be used.
+
+        closed : {'left', 'right', 'both', 'neither'}, optional
+            Whether the intervals are closed on the left-side, right-side, both
+            or neither.  If None, the existing closed will be used.
+        """
+        if left is None:
+
+            # no values passed
+            left, right = self.left, self.right
+
+        elif right is None:
+
+            # only single value passed, could be an IntervalArray
+            # or array of Intervals
+            if not isinstance(left, (type(self), ABCIntervalIndex)):
+                left = type(self)(left)
+
+            left, right = left.left, left.right
+        else:
+
+            # both left and right are values
+            pass
+
+        closed = closed or self.closed
+        return self._simple_new(
+            left, right, closed=closed, verify_integrity=False)
+
+    def copy(self, deep=False):
+        """
+        Return a copy of the array.
+
+        Parameters
+        ----------
+        deep : bool, default False
+            Also copy the underlying data backing this array.
+
+        Returns
+        -------
+        IntervalArray
+        """
+        left = self.left.copy(deep=True) if deep else self.left
+        right = self.right.copy(deep=True) if deep else self.right
+        closed = self.closed
+        # TODO: Could skip verify_integrity here.
+        return type(self).from_arrays(left, right, closed=closed)
+
+    def _formatting_values(self):
+        return np.asarray(self)
+
+    def isna(self):
+        return isna(self.left)
+
+    @property
+    def nbytes(self):
+        return self.left.nbytes + self.right.nbytes
+
+    @property
+    def size(self):
+        # Avoid materializing self.values
+        return self.left.size
+
+    @property
+    def shape(self):
+        return self.left.shape
+
+    @property
+    def itemsize(self):
+        return self.left.itemsize + self.right.itemsize
```

Comment:
@jschendel a bit after the fact, but I don't think that we should add this to `IntervalArray`. We deprecated `itemsize` some time ago for Series/Index (https://github.com/pandas-dev/pandas/pull/20721). 
Just noticed this because we get the warning during the test:

```
pandas/tests/indexes/interval/test_interval.py::TestIntervalIndex::()::test_itemsize
  /home/travis/build/pandas-dev/pandas/pandas/core/arrays/interval.py:694: FutureWarning: Int64Index.itemsize is deprecated and will be removed in a future version
    return self.left.itemsize + self.right.itemsize
```

Reply:
This was originally part of `IntervalIndex` and is currently what `IntervalIndex` calls under the hood, so if we remove it from `IntervalArray` we'd need to add it back `IntervalIndex`.  Can certainly do that, unless there's some objection based on it making the two inconsistent.  Removing it from `IntervalArray` seems like the best course of action though.

Reply:
Yes, removing it from `IntervalArray` is the best way forward I think. 
Also, on `IntervalIndex` it should be deprecated (now it is deprecated on the base IndexOpsMixin class, but by overriding it in IntervalIndex, it lost that deprecation. But apparently IntervalIndex is the only index that is not tested in the base tests, so this 'regression' was not catched by the tests).

Reply:
Opened https://github.com/pandas-dev/pandas/issues/22049 so we don't loose track of this discussion